packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -410,14 +417,14 @@ public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
 
       @Override
       public void onActivityStarted(Activity activity) {
-        if (!resolvingError && activity == Delegate.this.activity && googleApiClient != null) {
+        if (!resolvingError && activity == mRegistrar.activity() && googleApiClient != null) {
```

Comment:
Did semantics change here? If the `PluginRegistrar`'s activity can change dynamically, it might not be the same instance it had when the `Delegate` was created.

Similarly below.

Reply:
The PluginRegistrar's activity can change when the foreground activity is destroyed and recreated. FlutterActivityDelegate's onCreate will finish before the Delegate's onActivityStarted is called, so mRegistrar.activity() will already be the right activity. Likewise, the Delegate's onActivityStopped will be called before the foreground activity is destroyed.

It's also probalby worth noting here that googleApiClient is now created with the application's context rather than the foreground activity's context.